src/components/People.js

```diff
@@ -0,0 +1,78 @@
+import { Component } from "react";
+
+class People extends Component {
+  constructor() {
+    super();
+    this.state = {
+      characterData: [],
+      userInput: "",
+      display: false,
+      selectedCharcater: "",
+    };
+  }
+
+  handlefetch = () => {
+    fetch("https://ghibliapi.herokuapp.com/people")
+      .then((res) => {
+        return res.json();
+      })
+      .then((data) => {
+        console.log(data);
+        this.setState({
+          characterData: data,
+        });
+      });
+  };
+
+  handleUserInput = (e) => {
+    this.setState({
+      userInput: e.target.value,
+    });
+  };
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    let characterMatch = this.state.characterData.find((character) => {
+      return character.name === this.state.userInput;
+    });
+    this.setState({
+      userInput: characterMatch,
+      display: true,
+    });
+  };
```

Comment:
```suggestion
  handleSubmit = (e) => {
    e.preventDefault();
    let characterMatch = this.state.characterData.find((character) => {
      return character.name === this.state.userInput;
    });
    this.setState({
      selectedCharacter: characterMatch,
      display: characterMatch ? true : false,
    });
  };
```
There are similar problems here to your `handleChange` function in Movies.js.

If .find is unsuccessful in finding a match, it returns undefined.

This is the reason you're getting an error that says "cannot read property 'name' of undefined". With the way your code is written currently, if the user's search is unsuccessful, `this.state.display` will still be true, but `this.state.selectedCharacter` will be _undefined_, and trying to access `undefined.name` will throw an error. You can safeguard against this by making sure that `display` is **false** if `selectedCharacter` is _undefined_.

Again, this is just one way of fixing this problem. You don't actually need to keep `display` in state at all. In fact, I would recommend using `selectedCharacter` instead to determine if there is character info to display.